vkubelet/node.go

```diff
@@ -233,7 +233,7 @@ func (n *Node) controlLoop(ctx context.Context) error {
 	for {
 		select {
 		case <-ctx.Done():
-			return nil
+			return n.nodes.Delete(n.n.Name, &metav1.DeleteOptions{})
```

Comment:
This seems problematic. What if there are still running pods? Does normal kubelet exhibit this behaviour?